src/persist/src/operators/stream.rs

```diff
@@ -174,50 +175,23 @@ where
 
 /// Extension trait for [`Stream`].
 pub trait Seal<G: Scope<Timestamp = u64>, D: TimelyData> {
-    /// Passes through each element of the stream and seals the given collection (the `write`
-    /// handle) when the input frontier advances.
+    /// Passes through each element of the stream and calls
+    /// [`seal_all`](MultiWriteHandle::seal_all) on the given [`MultiWriteHandle`] when the input
+    /// frontier advances.
     ///
     /// This does not wait for the seal to succeed before passing through the data. We do, however,
     /// wait for the seal to be successful before allowing the frontier to advance. In other words,
     /// this operator is holding on to capabilities as long as seals corresponding to their
     /// timestamp are not done.
-    fn seal<K, V>(&self, name: &str, write: StreamWriteHandle<K, V>) -> Stream<G, (D, u64, isize)>
-    where
-        K: Codec,
-        V: Codec;
-
-    /// Passes through each element of the stream and seals the given primary and condition
-    /// collections, respectively, when their frontier advances. The primary collection is only
-    /// sealed up to a time `t` when the condition collection has also been sealed up to `t`.
-    ///
-    /// This does not wait for the seals to succeed before passing through the data. We do,
-    /// however, wait for the seals to be successful before allowing the frontier to advance. In
-    /// other words, this operator is holding on to capabilities as long as seals corresponding to
-    /// their timestamp are not done.
-    fn conditional_seal<D2: TimelyData, K, V, K2, V2>(
-        &self,
-        name: &str,
-        condition_input: &Stream<G, (D2, u64, isize)>,
-        primary_write: StreamWriteHandle<K, V>,
-        condition_write: StreamWriteHandle<K2, V2>,
-    ) -> Stream<G, (D, u64, isize)>
-    where
-        K: Codec,
-        V: Codec,
-        K2: TimelyData + Codec,
-        V2: TimelyData + Codec;
+    fn seal(&self, name: &str, write: MultiWriteHandle) -> Stream<G, (D, u64, isize)>;
```

Comment:
I think this could use a little bit more documentation on how to correctly use it - it seems like you have to do the flat map trick in order for this operator to work the way one might expect (otherwise it will seal everything to the frontier of one of the collections which seems surprising to me)

Reply:
Will do!